tests/foreman/api/test_contentviewversion.py

```diff
@@ -310,6 +320,595 @@ def test_negative_delete(self):
         # Make sure that content view version is still present
         self.assertEqual(len(content_view.read().version), 1)
 
+    @run_only_on('sat')
+    @tier2
+    def test_positive_remove_renamed_cv_version_from_default_env(self):
+        """Remove version of renamed content view from Library environment
+
+        @id: 7d5961d0-6a9a-4610-979e-cbc4ddbc50ca
+
+        @Steps:
+
+        1. Create a content view
+        2. Add a yum repo to the content view
+        3. Publish the content view
+        4. Rename the content view
+        5. remove the published version from Library environment
+
+        @Assert: content view version is removed from Library environment
+
+        @CaseLevel: Integration
+        """
+        new_name = gen_string('alpha')
+        org = entities.Organization().create()
+        # create yum product and repo
+        product = entities.Product(organization=org).create()
+        yum_repo = entities.Repository(
+            url=FAKE_1_YUM_REPO,
+            product=product,
+        ).create()
+        yum_repo.sync()
+        # create a content view and add to it the yum repo
+        content_view = entities.ContentView(organization=org).create()
+        content_view.repository = [yum_repo]
+        content_view = content_view.update(['repository'])
+        # publish the content view
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 1)
+        content_view_version = content_view.version[0].read()
+        self.assertEqual(len(content_view_version.environment), 1)
+        lce_library = entities.LifecycleEnvironment(
+            id=content_view_version.environment[0].id
+        ).read()
+        # ensure that the content view version is promoted to the Library
+        # lifecycle environment
+        self.assertEqual(lce_library.name, ENVIRONMENT)
+        # rename the content view
+        content_view.name = new_name
+        content_view.update(['name'])
+        self.assertEqual(content_view.name, new_name)
+        # delete the content view version from Library environment
+        content_view.delete_from_environment(lce_library.id)
+        # assert that the content view version does not exist in Library
+        # environment
+        self.assertEqual(len(content_view_version.read().environment), 0)
+
+    @run_only_on('sat')
+    @tier2
+    def test_positive_remove_qe_promoted_cv_version_from_default_env(self):
+        """Remove QE promoted content view version from Library environment
+
```

Comment:
As CV names are arbitrary, I'm not sure if we should include them in test names. I think the distinctive thing about this test is that it syncs docker content, I'd mention it at least in the description. How about test_positive_remove_promoted_docker_cv_version_from_default_env?